grep-0001-coding-guidelines.md

```diff
@@ -0,0 +1,122 @@
+# GREP [XXXX] -- [Title]
+
+[Note: All parts in square brackets are meant to be replaced as part of
+authoring a new GREP]
+
+- Original Author: Martin Braun <martin@gnuradio.org>,
+                   Marcus Müller <marcus@hostalia.de>
+- Champion: Marcus Müller <marcus@hostalia.de>
+- Status: Draft
+
+History:
+- 12-Feb-2018: Initial Draft
+
+## Abstract
+
+This document establishes coding guidelines for the GNU Radio main source tree.
+
+As a recommendation, these coding guidelines should also be adopted by
+out-of-tree modules.
+
+## Copyright / License
+
+This GREP is licensed as CC-BY-ND.
+Copyright 2018 The GNU Radio Foundation.
+
+## Motivation
+
+Common coding guidelines are the grease that makes a software project's
+cogwheels spin freely. They remove friction, and make certain unproductive
+discussions unnecessary. In particular because coding styles are subjective, it
+helps a great deal to simply declare one coding style as the one.
+
+New code should follow these rules, but there are always exceptions, and it is
+up to the individual contributor to decide what works best. That said, certain
+things such as indentation or whitespace rules can be taken care of by one's
+editor, and there is no reason to not adopt these settings. This makes it
+easier for multiple people to contribute to the same codebase, and will keep
+commits free of non-functional changes.
+
+To summarize, consistency is extremely useful, and coding guidelines help ensure
+consistency. We assume that all contributors are mature enough to at least try
+and follow them.
+
+## Description
+
+### General Coding Guidelines
+
+* Never submit code with trailing whitespace.
+* Code is read more often than it's written. Code readability is thus something
+  worth optimizing for.
+* Try and keep line lengths to 79 characters, unless readability suffers. We
```

Comment:
Really ?  79 ?
I mean it's 2018 and I can have side by side terminal with 140 char each ... now I agree my case might be extreme, but wouldn't taking a more modern value like 100 char width be more future looking ? Especially for C++ where lines do tend to get a bit long ?

Reply:
yeah, 79 is definitely at the lower end of useful limits (makes a lot of sense for other things, but not especially C++ in my experience). With a two- to four spaces indention depth, I'd argue I usually can live with up to ca 110 characters per line without losing the feeling that things on that line should be on that line. But that's definitely the uppermost boundary for length of lines where I'd be willing to not complain as a maintainer – so, 100 sounds fine to me.